### PR TITLE
Change build to be performed by the remote server

### PR DESCRIPTION
**Pull request by nicolasiensen**
In this PR:
- Fix `font-awesome` compilation (removing `webpack` plugin `UglifyJsPlugin`)
- Set env variables dynamically in `webpack/prod.config.js`
- Change `basscss-sass` source to fix build
- Add `postinstall` directive into the `package.json` to build project after install
- Set node version to `0.12.4`
- Update deploy instructions

### CODE REVIEW

**Comment by lpirola** [comment]
:clap:

File: package.json [hunk 1]
```diff
@@ -40,12 +44,13 @@
...
   },
   "dependencies": {
+    "autoprefixer-loader": "^2.0.0",
     "autoprefixer-core": "^5.2.0",
     "babel": "^5.8.21",
     "babel-loader": "^5.3.2",
     "babel-plugin-typecheck": "^1.2.0",
     "babel-runtime": "^5.8.20",
-    "basscss-sass": "^1.0.2",
+    "basscss-sass": "nicolasiensen/basscss-sass#v1.0.2-branch",
```
Comment (leticia) [inline package.json:53]: Should we use latest version? It seems that annoying bug is fixed: https://github.com/basscss/basscss-sass/issues/6#issuecomment-183780817

Comment (nicolasiensen) [inline package.json:53]: The problem is that the latest version of basscss is 6.x and we implemented
the front-end using version 1.x. So if we change it to latest it's
guaranteed that the front-end will break :(

Comment (leticia) [inline package.json:53]: In that case, let's merge this and later I can make a branch just to update it and ensure styles don't break. :dancers:

**Pull request merged**